src/io.c

```diff
@@ -449,9 +468,14 @@ dispatch_io_create_with_path(dispatch_io_type_t type, const char *path,
 				break;
 			default:
 				if ((path_data->oflag & O_CREAT) &&
-						(*(path_data->path + path_data->pathlen - 1) != '/')) {
+						!is_separator(*(path_data->path + path_data->pathlen - 1))) {
 					// Check parent directory
-					char *c = strrchr(path_data->path, '/');
+					char *c = NULL;
+					for (char *i = path_data->path; *i != '\0'; i++) {
```

Comment:
This is a slight change in semantics: `strchr` rather than than `strrchr`.

Reply:
The loop doesn't break after it finds a separator.

Reply:
True, but, why go forward rather than backwards?

Reply:
Ah, I missed that we know the length of the path. I can reverse it then.